tests/test_exceptions.py

```diff
@@ -40,3 +40,44 @@ def test_get_cause_response_message__no_message():
     error = PutError()
     assert error.cause_response_message is None
 
+
+def test_transact_error__get_reason_list_from_message():
+    message = 'Transaction cancelled, please refer cancellation reasons for specific reasons [None, None, None, ConditionalCheckFailed]'
+    assert TransactError._get_reason_list_from_message(message) == ['None', 'None', 'None', 'ConditionalCheckFailed']
+
+
+def test_transact_error__get_reason_list_from_message__no_match():
+    assert TransactError._get_reason_list_from_message('nope') is None
+
+
+def test_transact_error__parse_cancel_reason():
+    error = TransactError(
+        transact_items=[{'TableName': 'toot'}, {'TableName': 'boot'}],
+        msg='TransactionCanceledException',
+        cause=ClientError(
+            error_response={
+                'Error': {
+                    'Code': 'TransactionCanceledException',
+                    'Message': 'Transaction cancelled, please refer cancellation reasons for specific reasons [None, ConditionalCheckFailed]'
+                }
+            },
+            operation_name='test'
+        )
+    )
+    assert error.cancel_reasons == [({'TableName': 'toot'}, 'None'), ({'TableName': 'boot'}, 'ConditionalCheckFailed')]
```

Comment:
should `'None'` be `None` here?

Reply:
It's `None` from the string generated by DynamoDB. Do you suggest we treat `'None'` specially?

Reply:
yeah, probably makes the most sense to go through and replace all the `"None"`s with `None`